src/index.ts

```diff
@@ -17,11 +17,27 @@ export const angularModule = angular.module(name, [])
 
         init: function($scope, el): void {
           if (!el.hasAttribute('popgun-listening')) {
-            el.addEventListener('PopgunContentSetup', function(e) {
-              let pop = popgun.getPopFromGroupId((<Element>e.target).getAttribute('popgun-group'));
-              $compile(pop.popOver.element)($scope);
-              $scope.$apply();
-            }, false);
+            el.addEventListener('PopgunContentSetup', (function($compileScope) {
+              return function (e) {
+                  let $newScope = $compileScope.$new();
+                  let pop =
+                      popgun.getPopFromGroupId(e.target.getAttribute('popgun-group'));
+                  $compile(pop.popOver.element)($newScope);
+                  $newScope.$apply();
+              };
+            })($scope), false);
+            
+            var cleanupPopHandler = function (e) {
+                var pop = e.detail.pop;
+                if (pop) {
+                    angular.element(pop.popOver.element).scope().$destroy();
+                    angular.element(pop.popOver.element).remove();
+                }
+            };
+
+            el.addEventListener('PopgunContentSwap', cleanupPopHandler);
+            el.addEventListener('PopgunRemoveContent', cleanupPopHandler);
```

Comment:
Can we call this `PopgunContentRemove` to follow the naming convention of the other states?